packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -228,23 +228,13 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
 
           // Scan changed files only for incremental rebuilds.
           else if (rebuildStrategy === 'incremental') {
-            // TODO: use `scanDirResult.scanFiles(changedFiles)` for incremental
-            // rebuilds to prevent scanning the entire directory.
-
-            // Re-scan the directory to get the new `candidates`
-            scanDirResult = scanDir({
-              base, // Root directory, mainly used for auto content detection
-              contentPaths: compiler.globs.map((glob) => ({
-                base: inputBasePath, // Globs are relative to the input.css file
-                glob,
-              })),
-            })
+            let candidates = scanDirResult.scanFiles(changedFiles)
```

Comment:
Nice! This properly feels like "something we need to fix on the rust side" now :)